lib/nokogiri/css/parser_extras.rb

```diff
@@ -32,33 +38,22 @@ def clear_cache
         end
 
         # Execute +block+ without cache
-        def without_cache &block
-          tmp = @cache_on
-          @cache_on = false
+        def without_cache(&block)
+          tmp = cache_on?
+          set_cache false
           block.call
-          @cache_on = tmp
-        end
-
-        ###
-        # Parse this CSS selector in +selector+.  Returns an AST.
-        def parse selector
-          @warned ||= false
-          unless @warned
-            $stderr.puts('Nokogiri::CSS::Parser.parse is deprecated, call Nokogiri::CSS.parse(), this will be removed August 1st or version 1.4.0 (whichever is first)')
-            @warned = true
-          end
-          new.parse selector
+          set_cache tmp
```

Comment:
I think it would be good to have this line in `ensure`, so even if the block raises the state is restored correctly.

Reply:
Right on, thanks.

Reply:
See e2c694f